### PR TITLE
fix: Raise exception when meltano.yml is empty

### PR DESCRIPTION
**Pull request by rawwar**
When meltano.yml is empty, the following error is raised 
```
Traceback (most recent call last):
  File "/home/kalyan/anaconda3/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/kalyan/anaconda3/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/kalyan/.vscode-server/extensions/ms-python.python-2022.12.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher/../../debugpy/__main__.py", line 39, in <module>
    cli.main()
  File "/home/kalyan/.vscode-server/extensions/ms-python.python-2022.12.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/cli.py", line 430, in main
    run()
  File "/home/kalyan/.vscode-server/extensions/ms-python.python-2022.12.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/cli.py", line 284, in run_file
    runpy.run_path(target, run_name="__main__")
  File "/home/kalyan/.vscode-server/extensions/ms-python.python-2022.12.0/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 321, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/home/kalyan/.vscode-server/extensions/ms-python.python-2022.12.0/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 135, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/home/kalyan/.vscode-server/extensions/ms-python.python-2022.12.0/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 124, in _run_code
    exec(code, run_globals)
  File "/home/kalyan/work/meltano_project/meltano_debug/meltano/src/tester.py", line 8, in <module>
    cli(["environment",'list'])
  File "/home/kalyan/.cache/pypoetry/virtualenvs/meltano-P8gWqx08-py3.9/lib/python3.9/site-packages/click/core.py", line 1129, in __call__
    return self.main(*args, **kwargs)
  File "/home/kalyan/work/meltano_project/meltano_debug/meltano/src/meltano/cli/cli.py", line 39, in main
    return super().main(*args, windows_expand_args=False, **kwargs)
  File "/home/kalyan/.cache/pypoetry/virtualenvs/meltano-P8gWqx08-py3.9/lib/python3.9/site-packages/click/core.py", line 1054, in main
    rv = self.invoke(ctx)
  File "/home/kalyan/work/meltano_project/meltano_debug/meltano/src/meltano/cli/utils.py", line 541, in invoke
    super().invoke(ctx)  # noqa: WPS608
  File "/home/kalyan/.cache/pypoetry/virtualenvs/meltano-P8gWqx08-py3.9/lib/python3.9/site-packages/click/core.py", line 1653, in invoke
    super().invoke(ctx)
  File "/home/kalyan/.cache/pypoetry/virtualenvs/meltano-P8gWqx08-py3.9/lib/python3.9/site-packages/click/core.py", line 1403, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/kalyan/.cache/pypoetry/virtualenvs/meltano-P8gWqx08-py3.9/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/kalyan/.cache/pypoetry/virtualenvs/meltano-P8gWqx08-py3.9/lib/python3.9/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/kalyan/work/meltano_project/meltano_debug/meltano/src/meltano/cli/cli.py", line 84, in cli
    project = Project.find()
  File "/home/kalyan/.cache/pypoetry/virtualenvs/meltano-P8gWqx08-py3.9/lib/python3.9/site-packages/fasteners/lock.py", line 286, in wrapper
    return f(self, *args, **kwargs)
  File "/home/kalyan/work/meltano_project/meltano_debug/meltano/src/meltano/core/project.py", line 206, in find
    cls.activate(project)
  File "/home/kalyan/.cache/pypoetry/virtualenvs/meltano-P8gWqx08-py3.9/lib/python3.9/site-packages/fasteners/lock.py", line 286, in wrapper
    return f(self, *args, **kwargs)
  File "/home/kalyan/work/meltano_project/meltano_debug/meltano/src/meltano/core/project.py", line 133, in activate
    project.ensure_compatible()
  File "/home/kalyan/work/meltano_project/meltano_debug/meltano/src/meltano/core/behavior/versioned.py", line 33, in ensure_compatible
    file_version = self.file_version
  File "/home/kalyan/work/meltano_project/meltano_debug/meltano/src/meltano/core/project.py", line 167, in file_version
    return self.meltano.version
  File "/home/kalyan/work/meltano_project/meltano_debug/meltano/src/meltano/core/project.py", line 242, in meltano
    meltano_ff.get(f"{FEATURE_FLAG_PREFIX}.{FeatureFlags.ENABLE_UVICORN}")
AttributeError: 'NoneType' object has no attribute 'get'
```

This PR intends to catch any errors when meltano.yml is empty

### CODE REVIEW

**Comment by pandemicsyn** [comment]
Nice! @rawwar would you mind adding a test for this to make sure we catch any future regressions of this? 

(Probably in `tests/meltano/core/test_project.py` or maybe `tests/meltano/cli/test_cli.py` if you want to go for a more functional test)

**Comment by rawwar** [comment]
I managed to write test case. But, honestly, was not sure where to put `CLI_COMMANDS`. I did think of to put it in `tests/fixtures/cli.py` or `tests/conftest.py`.  I just put it in `tests/meltano/cli/test_cli.py` to request feedback.

**Comment by pandemicsyn** [comment]
> maybe there's a better way to do this?

You probably don't need to test with every command, anything that comes in via the main `cli` click group will trigger this I think. So I think you could do like the other commands [in test_cli.py and just invoke a single command ](https://github.com/meltano/meltano/blob/main/tests/meltano/cli/test_cli.py#L77) to exercise the path.

File: tests/meltano/cli/test_cli.py [hunk 1]
```diff
@@ -38,6 +75,14 @@ def test_activate_project(self, project, cli_runner, pushd):
         assert Project._default.root == project.root
         assert Project._default.readonly is False
 
+    @pytest.mark.parametrize("cli_command", CLI_COMMANDS)
+    def test_empty_meltanp_yml_project(
```
Comment (pandemicsyn) [inline tests/meltano/cli/test_cli.py:79]:
```suggestion
    def test_empty_meltano_yml_project(
```

Comment (pandemicsyn) [reply tests/meltano/cli/test_cli.py:79]: tiny typo

**Review by edgarrmondragon — APPROVED** [review]

**Comment by rawwar** [comment]
> You probably don't need to test with every command, anything that comes in via the main `cli` click group will trigger this I think.

Sure. I'll run it only for config command

**Review by edgarrmondragon — APPROVED** [review]
@rawwar lgtm! Checks are 🟢 so I'll defer to @pandemicsyn to give his final stamp and merge 😄.

**Review by pandemicsyn — APPROVED** [review]
lgtm! Thanks for the contrib @rawwar!

**Pull request merged**